### PR TITLE
Spacy removal

### PR DESCRIPTION
**Pull request by dcaustin33**
Remove spacy from dq

### CODE REVIEW

File: dataquality/__init__.py [hunk 1]
```diff
@@ -8,8 +8,8 @@
 :param val_data: Optional: validation data
 :param labels: The labels for the run
 :param framework: The framework to use, if provided it will be used instead of
-    inferring it from the model. For example, if you have a spacy model, you
-    can pass framework="spacy". If you have a torch model, you can pass
+    inferring it from the model. For example, if you have a pytorch model, you
+    can pass framework="pytorch". If you have a torch model, you can pass
     framework="torch"
```
Comment (setu4993) [inline dataquality/__init__.py:13]: `"torch"` or `"pytorch"`?

File: dataquality/dq_start/__init__.py [hunk 2]
```diff
@@ -320,8 +270,8 @@ def __init__(
         :param val_data: Optional: validation data
         :param labels: The labels for the run
         :param framework: The framework to use, if provided it will be used instead of
-            inferring it from the model. For example, if you have a spacy model, you
-            can pass framework="spacy". If you have a torch model, you can pass
+            inferring it from the model. For example, if you have a pytorch model, you
+            can pass framework="pytorch". If you have a torch model, you can pass
             framework="torch"
```
Comment (setu4993) [inline dataquality/dq_start/__init__.py:275]: Same ^.

File: docs/autodocs/_build/markdown/index.md [hunk 3]
```diff
@@ -79,15 +79,6 @@
...
-* [dataquality.integrations.spacy](api/dataquality.md#dataquality-integrations-spacy)
-
-
-    * [`watch()`](api/dataquality.md#dataquality.integrations.spacy.watch)
-
-
-    * [`unwatch()`](api/dataquality.md#dataquality.integrations.spacy.unwatch)
-
-
 * [dataquality.integrations.hf](api/dataquality.md#dataquality-integrations-hf)
```
Comment (setu4993) [inline docs/autodocs/_build/markdown/index.md:82]: Should we be modifying auto-docs?

File: pyproject.toml [hunk 4]
```diff
@@ -85,7 +84,7 @@ test = [
     "setfit",
     "accelerate>=0.19.0",
     "typing-inspect==0.8.0",
-    "typing-extensions<4.6.0",
+    "typing-extensions==4.0.0",
```
Comment (setu4993) [inline pyproject.toml:87]: Probably better:

```suggestion
    "typing-extensions>=4.0.0,<5.0.0",
```

**Review by setu4993 — COMMENTED** [review]
So much 🔥!

1 question about `"torch"` vs. `"pytorch"` for framework name. Otherwise looks good!

File: pyproject.toml (hunk 4 above)
Comment (dcaustin33) [reply pyproject.toml:87]: It actually does not work with anything other than 4.0 rn lol check the other commits I tried 4.3 4.2 4.1

Comment (setu4993) [reply pyproject.toml:87]: So weird. Thanks for checking!

Comment (elboy3) [reply pyproject.toml:87]: in what way is it failing?

Comment (dcaustin33) [reply pyproject.toml:87]: So weird will have to get to that eventually!

**Review by elboy3 — APPROVED** [review]
Before merging let's try to get it working with other versions of typing_extensions

Also I'd check in with franz about those autodocs and editing those manually

File: pyproject.toml (hunk 4 above)
Comment (dcaustin33) [reply pyproject.toml:87]: 46 different errors check the commit history above

File: docs/autodocs/_build/markdown/index.md (hunk 3 above)
Comment (dcaustin33) [reply docs/autodocs/_build/markdown/index.md:82]: I can revert - did not mean to do this was just deleting everything with spacy in it

File: pyproject.toml (hunk 4 above)
Comment (setu4993) [reply pyproject.toml:87]: `typing_extensions` causing failures is super weird.

File: docs/autodocs/_build/markdown/index.md (hunk 3 above)
Comment (setu4993) [reply docs/autodocs/_build/markdown/index.md:82]: I don't know if it'll actually cause an issue 😅. Let's check with @franz101?

**Comment by anthonycorletti** [comment]
Au revoir

**Pull request merged**